### ISSUE TITLE
Understanding batching with videos in temporal attention network

### ISSUE THREAD

**Issue by CarmelaCalabrese** [wengzejia1/Open-VCLIP#11]
Hi @wengzejia1 ,
sorry again. I am using your code combined with another architecture on top of CLIP but I would need to understand how you feed video in your temporal attention network. When you call the forward function of the class MultiheadAttention, you check:

is_batched = query.dim() == 3
if self.batch_first and is_batched:

Could you kindly explain these two boolean variables and, in general, how you process video data from the dataloader to the forward function?

Thank you in advance!

**Comment by wengzejia1** [comment]
Hi. The class MultiheadAttention(Module) mentioned in your question comes from the pytorch source code (we modified the attention operation based on the source code). The two variables you mentioned (``is_batched'' and ``batch_first'') are used in the source code to determine whether it is a batch input and to determine the order of each dimension. You can see the explanation of these two variables in the code comments.

```
Args:
        query: Query embeddings of shape :math:`(L, E_q)` for unbatched input, :math:`(L, N, E_q)` when ``batch_first=False``
            or :math:`(N, L, E_q)` when ``batch_first=True``, where :math:`L` is the target sequence length,
            :math:`N` is the batch size, and :math:`E_q` is the query embedding dimension ``embed_dim``.
            Queries are compared against key-value pairs to produce the output.
```
